src/args.rs

```diff
@@ -104,6 +111,19 @@ pub struct CliArgs {
     #[clap(short = 'u', long = "upload-files")]
     pub file_upload: bool,
 
+    /// Specify uploadable media types
+    #[clap(arg_enum, short = 'm', long = "media-type", requires = "file-upload")]
+    pub media_type: Option<Vec<MediaType>>,
+
+    /// Directly specify the uploadable media type expression
+    #[clap(
+        short = 'M',
+        long = "raw-media-type",
+        requires = "file-upload",
+        conflicts_with = "media-type"
+    )]
+    pub media_type_raw: Option<String>,
```

Comment:
Is this really useful? At the point where you need this level of control, wouldn't it be better to just generally set a header using `--header`?

Reply:
How would you achieve the same thing using a header?

I think the `raw-media-type` parameter can be useful if you want to specify a filetype or a media type that is not included in the `media-type` parameter.
There are a lot of possibilities for browsers to use this value and offer the right file choosing option to the user, and I think by adding this option we gain a lot of flexibility without adding a lot of complexity.

To give an example, if you wanted to upload a PDF document, you could use `miniserve -u -M application/pdf` and your file chooser would automatically filter for PDF files.
On stock Android, this results in a nice overview of your most recent PDF files all in one place, allowing to quickly find your file.

Reply:
Nevermind, of course this can't be accomplished using header. I really like your explanation. Merging as is!